src/app/mapController.js

```diff
@@ -1,18 +1,114 @@
 define([
-    'agrc/widgets/map/BaseMap'
+    'agrc/widgets/map/BaseMap',
+
+    'app/config',
+    'app/router',
+
+    'dojo/_base/lang',
+    'dojo/Deferred',
+    'dojo/topic',
+    'dojo/when',
+
+    'esri/geometry/Extent',
+    'esri/layers/FeatureLayer',
+    'esri/tasks/query'
 ], function (
-    BaseMap
+    BaseMap,
+
+    config,
+    router,
+
+    lang,
+    Deferred,
+    topic,
+    when,
+
+    Extent,
+    FeatureLayer,
+    Query
 ) {
     return {
+        // layers: FeatureLayer[]
+        //      POINT, LINE & POLY layers
+        layers: [],
+
         initMap: function (mapDiv) {
             // summary:
-            //      Sets up the map
-            console.info('app.App::initMap', arguments);
+            //      Sets up the map and layers
+            console.info('app/mapController/initMap', arguments);
 
-            this.map = new BaseMap(mapDiv, {
+            var that = this;
+            that.map = new BaseMap(mapDiv, {
                 showAttribution: false,
                 defaultBaseMap: "Hybrid"
             });
+
+            // suspend base map layer until we get the initial extent
+            // trying to save requests to the server
+            var baseLayer = that.map.getLayer(that.map.layerIds[0]);
+            baseLayer.suspend();
+
+            that.addLayers();
+
+            that.map.on('load', function () {
+                that.selectLayers().then(lang.hitch(baseLayer, 'resume'));
+            });
+
+            topic.subscribe(config.topics.projectIdsChanged, lang.hitch(that, 'selectLayers'));
+        },
+        selectLayers: function () {
+            // summary:
+            //      selects the feature layers based upon the current project ids
+            // returns: Promise
+            console.log('app/mapController:selectLayers', arguments);
+
+            var def = new Deferred();
+            var q = new Query();
+            q.where = router.getProjectsWhereClause();
+            // don't show all features for feature layers
+            if (q.where !== '1 = 1') {
+                this.layers.forEach(function (lyr) {
+                    lyr.selectFeatures(q);
+                });
+                var that = this;
+                when(router.getProjectIdsExtent(), function (extent) {
```

Comment:
this is really cool, but I think it would be better to save the 3 extra requests to the server and use the layers return object off of the selection event.